Mathlib/Algebra/BigOperators/Group/List/Basic.lean

```diff
@@ -223,6 +223,18 @@ end Monoid
 section CommMonoid
 variable [CommMonoid M] {a : M} {l l₁ l₂ : List M}
 
+@[to_additive (attr := simp)]
+theorem prod_insertIdx {i} (h : i ≤ l.length) : (l.insertIdx i a).prod = a * l.prod := by
```

Comment:
Could you add the non-commutative version too, in terms of `take` and `drop`?

Reply:
OK no problem.

Reply:
Hi, I checked and found that `prod_take_mul_prod_drop` already exists. Could you clarify on what other propositions you mean by this?

Reply:
Apologies, I'd missed that.

Perhaps it would be useful to modify this lemma to assume `∀ a' ∈ l.take i, Commute a a'`? You could then have another version that moves `a` to the end, expressed with `drop`